src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/DocumentPullDiagnosticsEndpoint.cs

```diff
@@ -42,6 +42,12 @@ internal class DocumentPullDiagnosticsEndpoint : IRazorRequestHandler<VSInternal
     public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, VSInternalClientCapabilities clientCapabilities)
     {
         serverCapabilities.SupportsDiagnosticRequests = true;
+        if (serverCapabilities.DiagnosticProvider is null)
+        {
+            serverCapabilities.DiagnosticProvider = new VSInternalDiagnosticOptions();
+        }
+
+        serverCapabilities.DiagnosticProvider.DiagnosticKinds = [ VSInternalDiagnosticKind.Syntax];
```

Comment:
Nit: The null check seems unnecessary. This could probably all just be:

```suggestion
        serverCapabilities.DiagnosticProvider = new VSInternalDiagnosticOptions()
        {
            DiagnosticKinds = [VSInternalDiagnosticKind.Syntax]
        }
```

Reply:
The ApplyCapabilities seems like a 'Decorator'; I'm avoiding the accidental overwrite of a property set elsewhere.  I have switced to using the ??= as proposed in another comment, so the method itself is now half the original length.